perfdash/perfdash.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	// TODO: move this somewhere central
+	"k8s.io/contrib/submit-queue/jenkins"
+	"k8s.io/kubernetes/pkg/util/sets"
+)
+
+type LatencyData struct {
+	APICalls []APICallLatency `json:"apicalls"`
+}
+
+type APICallLatency struct {
+	Latency  Histogram `json:"latency"`
+	Resource string    `json:"resource"`
+	Verb     string    `json:"verb"`
+}
+
+type Histogram map[string]float64
+
+type resourceToHistogram map[string][]APICallLatency
+
+var buildLatency = map[int]resourceToHistogram{}
+
+func main() {
+	job := "kubernetes-e2e-gce-scalability"
+	client := jenkins.JenkinsClient{
+		Host: "http://kubekins.dls.corp.google.com",
+	}
+
+	queue, err := client.GetJob(job)
+	if err != nil {
+		fmt.Printf("%v\n", err)
+		return
+	}
+	resources := sets.NewString()
+	for ix := range queue.Builds {
+		build := queue.Builds[ix]
+		reader, err := client.GetConsoleLog(job, build.Number)
+		if err != nil {
+			fmt.Printf("error getting logs: %v", err)
+			continue
+		}
+		defer reader.Close()
+		scanner := bufio.NewScanner(reader)
+		buff := &bytes.Buffer{}
+		inLatency := false
+
+		hist := resourceToHistogram{}
+		found := false
+		for scanner.Scan() {
+			line := scanner.Text()
+			// TODO: This is brittle, we should emit a tail delimiter too
+			if strings.Contains(line, "INFO") || strings.Contains(line, "STEP") || strings.Contains(line, "Failure") {
+				if inLatency {
+					obj := LatencyData{}
+					if err := json.Unmarshal(buff.Bytes(), &obj); err != nil {
+						fmt.Printf("error parsing JSON in build %d: %v %s\n", build.Number, err, buff.String())
+						return
```

Comment:
Do we want to return here?
I think that in case of corrupted data in some build, we should break everything - we should just break the computation for this particular build.